plasmapy/formulary/parameters.py

```diff
@@ -1198,7 +1198,7 @@ def gyroradius(
 
     # check 1: ensure either Vperp or T_i invalid, keeping in mind that
     # the underlying values of the astropy quantity may be numpy arrays
-    if np.any(np.logical_not(np.logical_xor(isfinite_Vperp, isfinite_Ti))):
+    if np.any(np.logical_and(isfinite_Vperp, isfinite_Ti)):
```

Comment:
Do you think we could test this to ensure it works as expected? The tests for gyroradius are located around https://github.com/PlasmaPy/PlasmaPy/blob/main/plasmapy/formulary/tests/test_parameters.py#L798 here :)

Reply:
Ahh, didn't knew about that proper test section, sorry - had only made tests on my own. Will do the proper testing later.

Reply:
Does _testing_ here simply means to run `test_gyroradius()`, or also to modify `test_gyroradius()` and add some new tests? (The former one should be sufficient in my opinion but I might overlook something)

Reply:
The tests are ran automatically anyway (see the Checks tab!), so that's already being done :) What I meant instead was to add (or ensure there are already! haven't checked 😅) a few checks to `test_gyroradius`:

1. gyroradius(T_i = nan) == nan, no exception;
2. gyroradius(Vperp = nan) == nan, no exception;
3. gyroradius(T_i = nan, Vperp = nan) raises exception (there should be a `pytest.raises` clause around that general area we could adapt :) )

(`np.isnan` would be helpful for the comparison, since anything `== np.nan` is `False`, even if that anything is `np.nan`!)

Reply:
Those are exactly the checks I have done manually :)

Now I will try to understand the syntax in the `test_gyroradius` function and include them there.

Reply:
Note that this might not be (well, I'm pretty sure it is not) the most elegant or "pythonic" way of testing it, but all the three new cases are included via `assert` tests. Not sure though if this qualifies the official test criteria....?

Reply:
This is literally what I'm here for :)

They're great, exactly what I was hoping for! Since the test suite isn't exploding, we can probably merge it now :)